src/executables/master_server_monitor.cpp

```diff
@@ -22,92 +22,353 @@
 #include <algorithm>
-
+#include <unistd.h>
+#include <signal.h>
+#include <system_parameter.h>
 #include "master_server_monitor.hpp"
 
 std::unique_ptr<server_monitor> master_Server_monitor = nullptr;
 
 server_monitor::server_monitor ()
 {
-  m_server_entry_list = std::make_unique<std::list<server_entry>> ();
-  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_entry_list is created. \n");
+  // arbitrary size
+  constexpr int SERVER_MONITOR_JOB_QUEUE_SIZE = 1024;
+  m_job_queue = new lockfree::circular_queue<server_monitor_job> (SERVER_MONITOR_JOB_QUEUE_SIZE);
+  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : job_queue is created. \n");
 
-  m_thread_shutdown = false;
-  m_monitoring_thread = std::make_unique<std::thread> ([this]()
-  {
-    while (!m_thread_shutdown)
-      {
-	// TODO: select server_entry whose m_need_revive value is true. (Will be implemented in CBRD-25438 issue.)
-      }
-  });
+  m_server_entry_map = std::unordered_map<std::string, server_entry> ();
+  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_entry_map is created. \n");
+
+  start_monitoring_thread ();
 
-  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_monitor_thread is created. \n");
   fflush (stdout);
 }
 
 // In server_monitor destructor, it should guarentee that
 // m_monitoring_thread is terminated before m_monitor_list is deleted.
 server_monitor::~server_monitor ()
+{
+  assert (m_job_queue->is_empty ());
+  if (m_job_queue)
+    {
+      delete m_job_queue;
+      fprintf (stdout, "[SERVER_REVIVE_DEBUG] : job_queue is deleted. \n");
+    }
+
+  stop_monitoring_thread ();
+
+
+  assert (m_server_entry_map.size () == 0);
+  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_entry_map is deleted. \n");
+
+  fflush (stdout);
+}
+
+void
+server_monitor::start_monitoring_thread ()
+{
+  m_thread_shutdown = false;
+
+  m_monitoring_thread = std::make_unique<std::thread> (&server_monitor::server_monitor_thread_worker, this);
+  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_monitor_thread is created. \n");
+}
+
+void
+server_monitor::stop_monitoring_thread ()
 {
   m_thread_shutdown = true;
+
   if (m_monitoring_thread->joinable())
     {
+      m_monitor_cv_consumer.notify_all();
       m_monitoring_thread->join();
       fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_monitor_thread is terminated. \n");
     }
+}
 
-  assert (m_server_entry_list->size () == 0);
-  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : server_entry_list is deleted. \n");
-  fflush (stdout);
+void
+server_monitor::server_monitor_thread_worker ()
+{
+  while (!m_thread_shutdown)
+    {
+      std::unique_lock<std::mutex> lock (m_monitor_mutex_consumer);
+
+      constexpr int period_in_secs = 1;
+      while (!m_monitor_cv_consumer.wait_for (lock, std::chrono::seconds (period_in_secs), [this] ()
+      {
+	return !m_job_queue->is_empty () || m_thread_shutdown;
+	}));
+
+      while (!m_job_queue->is_empty ())
+	{
+	  server_monitor_job consume_job;
+	  m_job_queue->consume (consume_job);
```

Comment:
전반적인 코딩 스타일로 생각되는데, 빈라인을 추가해서 코드 가독성을 고려해 주세요.

1. 변수 선언 다음 라인
2. 주요 문맥 구분; 다음은 예시이며, 문맥에 따라 다를 수 있지만 고려
    - if 구문 시작 시
    - for, while 등 시작 시
    - switch 문 시작 시
    - return 전 후 등

```suggestion
	  server_monitor_job consume_job;
	  
	  m_job_queue->consume (consume_job);
```

Reply:
전체적으로 코드 검토하여 빈 라인 추가하였습니다.